mongomock/collection.py

```diff
@@ -254,6 +256,8 @@ def update(self, spec, document, upsert = False, manipulate = False,
                         existing_document.update(self._internalize_dict(document))
                         if existing_document['_id'] != _id:
```

Comment:
Yes, I think it should

On Tue Feb 17 2015 at 8:17:36 PM Marc Prewitt <notifications@github.com>
wrote:

> In mongomock/collection.py
> <https://github.com/vmalloc/mongomock/pull/129#discussion_r24838768>:
>
> > @@ -254,6 +256,8 @@ def update(self, spec, document, upsert = False, manipulate = False,
> >                          existing_document.update(self._internalize_dict(document))
> >                          if existing_document['_id'] != _id:
>
> @vmalloc <https://github.com/vmalloc> When I was fixing this to handle
> dict _id's I found that mongo doesn't allow changing _id in an update. It
> throws this message:
>
> The _id field cannot be changed from ...
>
> For compatibility, should this throw an exccption instead of delete/insert?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/vmalloc/mongomock/pull/129/files#r24838768>.
>